### PR TITLE
Feature/add roles localizations edit endpoints

### PR DESCRIPTION
**Pull request by ndoriomedov**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/942
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/946

### CODE REVIEW

File: src/RightsService.Business/Commands/RoleLocalization/CreateRoleLocalizationCommand.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    private readonly IAccessValidator _accessValidator;
+    private readonly IResponseCreater _responseCreator;
+    private readonly IRoleLocalizationRepository _roleLocalizationRepository;
+    private readonly IDbRoleLocalizationMapper _roleLocalizationMapper;
+    private readonly ICreateRoleLocalizationRequestValidator _requestValidator;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+
+    public CreateRoleLocalizationCommand(
+      IAccessValidator accessValidator,
+      IResponseCreater responseCreator,
```
Comment (MariyaSin) [inline src/RightsService.Business/Commands/RoleLocalization/CreateRoleLocalizationCommand.cs:31]: update kernel

File: src/RightsService.Business/Commands/RoleLocalization/CreateRoleLocalizationCommand.cs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    public async Task<OperationResultResponse<Guid?>> ExecuteAsync(CreateRoleLocalizationRequest request)
+    {
+      if (!await _accessValidator.IsAdminAsync())
+      {
+        return _responseCreator.CreateFailureResponse<Guid?>(HttpStatusCode.Forbidden);
+      }
+
+      if (!request.RoleId.HasValue)
```
Comment (MariyaSin) [inline src/RightsService.Business/Commands/RoleLocalization/CreateRoleLocalizationCommand.cs:52]: could you put the check to validator?

File: src/RightsService.Business/Commands/RoleLocalization/EditRoleLocalizationCommand.cs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        if (isActive
+          && await _roleLocalizationRepository.DoesLocaleExistAsync(roleLocalization.RoleId, roleLocalization.Locale))
+        {
+          return _responseCreator.CreateFailureResponse<bool>(
+            HttpStatusCode.BadRequest,
+            new List<string> { "Role must have only one localization per locale." });
+        }
+      }
+
+      if (nameOperation != default
```
Comment (MariyaSin) [inline src/RightsService.Business/Commands/RoleLocalization/EditRoleLocalizationCommand.cs:84]: use validator

File: src/RightsService.Business/Commands/RoleLocalization/EditRoleLocalizationCommand.cs [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        bool isActive = bool.Parse(isActiveOperation.value.ToString().Trim());
+
+        if (roleLocalization.IsActive == isActive)
+        {
+          return _responseCreator.CreateFailureResponse<bool>(
+            HttpStatusCode.BadRequest,
+            new List<string> { "Role localization already has this status." });
+        }
+
+        if (isActive
```
Comment (MariyaSin) [inline src/RightsService.Business/Commands/RoleLocalization/EditRoleLocalizationCommand.cs:75]: use validator

File: src/RightsService.Validation/EditRoleLocalizationRequestValidator.cs [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+      AddFailureForPropertyIf(
+        nameof(EditRoleLocalizationRequest.IsActive),
+        x => x == OperationType.Replace,
+        new Dictionary<Func<Operation<EditRoleLocalizationRequest>, bool>, string>
+        {
+          {
+            x => bool.TryParse(x.value?.ToString(), out _), "Incorrect isActive format."
+          }
+        },
+        CascadeMode.Stop);
```
Comment (MariyaSin) [inline src/RightsService.Validation/EditRoleLocalizationRequestValidator.cs:56]: remove it

**Review by MariyaSin — APPROVED** [review]

File: src/RightsService.Business/Commands/RoleLocalization/CreateRoleLocalizationCommand.cs (hunk 2 above)
Comment (ndoriomedov) [reply src/RightsService.Business/Commands/RoleLocalization/CreateRoleLocalizationCommand.cs:52]: can't get roleId in patch validator

File: src/RightsService.Business/Commands/RoleLocalization/EditRoleLocalizationCommand.cs (hunk 3 above)
Comment (ndoriomedov) [reply src/RightsService.Business/Commands/RoleLocalization/EditRoleLocalizationCommand.cs:84]: DoesNameExistAsync(string locale, string name) requires roleLocalization's locale, can't get it in validator

File: src/RightsService.Business/Commands/RoleLocalization/EditRoleLocalizationCommand.cs (hunk 4 above)
Comment (ndoriomedov) [reply src/RightsService.Business/Commands/RoleLocalization/EditRoleLocalizationCommand.cs:75]: i need to compare roleLocalization's status with status from the request, can't get it in validator

**Review by Cheephot — APPROVED** [review]

**Review by VladSpitsin — APPROVED** [review]

**Review by grinakate — APPROVED** [review]

**Pull request merged**